spark/src/test/java/com/netflix/iceberg/spark/source/TestDataFrameWrites.java

```diff
@@ -91,23 +98,55 @@ public static void stopSpark() {
 
   @Override
   protected void writeAndValidate(Schema schema) throws IOException {
+    writeAndValidateWithLocations(schema, false, false);
+  }
+
+  @Test
+  public void testWrite_overridingDataLocation_tablePropertyOnly() throws IOException {
+    writeAndValidateWithLocations(BASIC_SCHEMA, true, false);
+  }
+
+  @Test
+  public void testWrite_overridingDataLocation_sourceOptionOnly() throws IOException {
+    writeAndValidateWithLocations(BASIC_SCHEMA, false, true);
+  }
+
+  @Test
+  public void testWrite_overridingDataLocation_sourceOptionTakesPrecedence() throws IOException {
+    writeAndValidateWithLocations(BASIC_SCHEMA, true, true);
+  }
+
+  private void writeAndValidateWithLocations(
+      Schema schema,
+      boolean setTablePropertyDataLocation,
```

Comment:
Hm, rereading this I think it tries too hard to reuse code in exchange for the antipattern of using boolean switches. This can be written more idiomatically.